TheMET/TheMET/Network/NetworkManager.swift

```diff
@@ -9,7 +9,20 @@ import Foundation
 
 class NetworkManager {
     
+    private var lastRequestTime: Date = .now
+    
     func get(urlString: String, parameters: [String : String], completion: @escaping ([String: Any]?) -> Void)  {
+        if self.isTimeIntervalLongEnough(lastRequest: lastRequestTime) {
+            self.executeNetworkGet(urlString: urlString, parameters: parameters, completion: completion)
+        } else {
+            DispatchQueue.global().asyncAfter(deadline: DispatchTime.now() + .milliseconds(Int(12.5)), qos: .background) {
+                self.executeNetworkGet(urlString: urlString, parameters: parameters, completion: completion)
```

Comment:
It is better to pass a `weak` copy of `self` to this closure. So that the NetworkService could be deconstructed at any point of time. Otherwise this closure will prevent such deconstruction.